src/app/store/common/api-clients.hooks.ts

```diff
@@ -1,27 +1,42 @@
 import { useMemo } from 'react';
 
-import { apiClients, createConfig } from '@app/store/common/api-clients';
+import { BitcoinClient } from '@app/query/bitcoin/bitcoin-client';
+import { StacksClient } from '@app/query/stacks/stacks-client';
+import { createConfig } from '@app/query/stacks/utils';
+import {
+  BITCOIN_API_BASE_URL_MAINNET,
+  BITCOIN_API_BASE_URL_TESTNET,
+  DEFAULT_SERVER_MAINNET,
+} from '@shared/constants';
 
 import { useCurrentNetworkState } from '../networks/networks.hooks';
 
+export function useBitcoinClientState() {
+  const network = useCurrentNetworkState();
+
+  const baseUrl =
+    network.url === DEFAULT_SERVER_MAINNET
+      ? BITCOIN_API_BASE_URL_MAINNET
+      : BITCOIN_API_BASE_URL_TESTNET;
```

Comment:
`whenBitcoinNetwork` helper?

Reply:
In fact, as this is based on the STX network state, we could use `whenNetwork` helper